File: Tutoriels/PHP/z1_Make_Ecommerce_Site_Weekend/php/class/GSetting.php
<?php   
    class GSetting {
        //===============================================
        private static $m_instance = null;
		const G_CHAR_COMMENT = '#';
        //===============================================
        private function __construct() {
        
        }
        //===============================================
        public static function Instance() {
            if(is_null(self::$m_instance)) {
                self::$m_instance = new GSetting();  
            }
            return self::$m_instance;
        }
        //===============================================
        public function load($file) {
			$lFile = fopen($file, "r");
			
			while(!feof($lFile)) {
				$lLine = fgets($lFile);
				$lLine = GString::Instance()->trim($lLine);
				if($lLine == "") continue;
				if($lLine[0] == self::G_CHAR_COMMENT) continue;
				$lSplit = GString::Instance()->split($lLine);
				$lKey = $lSplit[0];
				$lValue = $lSplit[1];
				GConfig::Instance()->setData($lKey, $lValue);
			}
			//GConfig::Instance()->showData();
        }
        //===============================================
    }
?>

File: Tutoriels/Cpp/Qt/Projets/src/manager/GSetting.cpp
//================================================
#include "GSetting.h"
#include "GString.h"
#include "GPrint.h"
#include "GConfig.h"
//================================================
GSetting* GSetting::m_instance = 0;
//================================================
GSetting::GSetting() {

}
//================================================
GSetting::~GSetting() {

}
//================================================
GSetting* GSetting::Instance() {
    if(m_instance == 0) {
        m_instance = new GSetting;
    }
    return m_instance;
}
//================================================
void GSetting::load(const string& file) {
    ifstream lFile(file);
    string lLine;
    while(getline(lFile, lLine)) {
        lLine = GString::Instance()->trim(lLine);
        if(lLine == "") continue;
        if(lLine[0] == G_CHAR_COMMENT) continue;
        vector<string> lSplit = GString::Instance()->split(lLine, "=");
        string lKey = GString::Instance()->trim(lSplit[0]);
        string lValue = GString::Instance()->trim(lSplit[1]);
        GConfig::Instance()->setData(lKey, lValue);
    }
    lFile.close();
    GConfig::Instance()->showData();
}
//================================================
